atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleOverflowWriteTable.java

```diff
@@ -249,6 +306,58 @@ private void deleteOverflow(String overflowTable, List<Object[]> args) {
                 args);
     }
 
+    private void deleteOverflowRange(String overflowTable, String shortTableName, RangeRequest range) {
+        StringBuilder query = new StringBuilder();
+        query.append(" /* DELETE_RANGE_OVERFLOW (").append(overflowTable).append(") */ ");
+        query.append(" DELETE /*+ INDEX(m pk_").append(overflowTable).append(") */ ");
+        query.append("   FROM ").append(overflowTable).append(" m ");
+        query.append(" WHERE m.id IN (");
+
+        // subquery for finding rows in the short table
+        query.append("SELECT /*+ INDEX(i pk_").append(shortTableName).append(") */ ");
+        query.append("       i.overflow ");
+        query.append("    FROM ").append(shortTableName).append(" i ");
+
+        // add where clauses
+        byte[] start = range.getStartInclusive();
```

Comment:
This is also duplicated in 2 other places. Might need a stronger QueryBuilder that contains a `StringBuilder` and a list of `args`, with a method like `addWhereClauses(RangeRequest range)`.